static/app/views/onboarding/setupDocs.tsx

```diff
@@ -68,81 +68,163 @@ function MissingExampleWarning({
   );
 }
 
+function LoadGettingStartedDoc({
+  platform,
+  orgSlug,
+  projectSlug,
+  activeProductSelection,
+}: {
+  activeProductSelection: PRODUCT[];
+  orgSlug: Organization['slug'];
+  platform: PlatformIntegration | null;
+  projectSlug: Project['slug'];
+}) {
+  const [module, setModule] = useState<null | {
+    default: React.ComponentType<{activeProductSelection: PRODUCT[]; dsn: string}>;
+  }>(null);
+
+  const platformPath =
+    platform?.type === 'framework'
+      ? platform?.id.replace(`${platform.language}-`, `${platform.language}/`)
+      : platform?.id;
+
+  const {
+    data: projectKeys = [],
+    isError: projectKeysIsError,
+    isLoading: projectKeysIsLoading,
+  } = useApiQuery<ProjectKey[]>([`/projects/${orgSlug}/${projectSlug}/keys/`], {
+    staleTime: Infinity,
+  });
+
+  useEffect(() => {
+    if (projectKeysIsError || projectKeysIsLoading) {
+      return;
+    }
+
+    async function getGettingStartedDoc() {
+      const mod = await import(
+        /* webpackExclude: /.spec/ */
+        `sentry/gettingStartedDocs/${platformPath}`
+      );
+      setModule(mod);
+    }
+    getGettingStartedDoc();
+  }, [platformPath, projectKeysIsError, projectKeysIsLoading, projectKeys]);
+
+  if (!module) {
+    return null;
+  }
+
+  const {default: GettingStartedDoc} = module;
+  return (
+    <GettingStartedDoc
+      dsn={projectKeys[0].dsn.public}
+      activeProductSelection={activeProductSelection}
+    />
+  );
+}
+
 export function DocWithProductSelection({
   organization,
   location,
   projectSlug,
   newOrg,
-  currentPlatform,
+  currentPlatform: currentPlatformKey,
 }: {
   currentPlatform: PlatformKey;
   location: Location;
   organization: Organization;
   projectSlug: Project['slug'];
   newOrg?: boolean;
 }) {
+  const products = useMemo(
+    () => (location.query.product ?? []) as PRODUCT[],
+    [location.query.product]
+  );
+
```

Comment:
i don't think `useMemo` is being helpful here. I'd have to test to be sure, but I don't think the object identities are going to be stable

Reply:
I was getting a warning